internal/providers/docker/components/container/lifecycle.go

```diff
@@ -421,3 +470,129 @@ func (c *Container) removeExistingContainerByName(ctx context.Context) error {
 
 	return nil
 }
+
+func (c *Container) endpointSettings(sn compose.ServiceNetwork) *network.EndpointSettings {
+	es := &network.EndpointSettings{
+		Aliases: append([]string{c.ComponentID, c.ComponentName}, sn.Aliases...),
```

Comment:
Does docker-compose do something similar here or is this some aliasing logic we're adding for convenience?

What happens if there is a conflict on these aliases?

Reply:
Yes, this is essentially what docker-compose does as well, and we will use the same names as long as we have not mangled the name to fit the component naming convention. Since we find and replace any containers that Docker Compose created, there should not be any conflict.